tests/test_filequeue.py

```diff
@@ -67,11 +67,53 @@ async def testNoSleepFileQueue(self):
 
         os.link(self.tmp_file, os.path.join(self.tmp_dir, os.path.basename(self.tmp_file)))
 
+        file_list = await fileq.dequeue_files()
+        # there should be no files to dequeue, because the directory scanner
+        # hasn't had a chance to run.
+        self.assertEqual(len(file_list), 0)
+        await asyncio.sleep(3)
```

Comment:
Earlier (line 53) the sleep interval was 1 s, here (and later) it is set to 3 s. Is it necessary? If it is, a comment briefly explaining why it is so would be nice. If it is not, I'd use the same value everywhere. BTW, in the latter case, you could use a named constant instead to avoid these "magic numbers".

Reply:
It's intended to keep the file scanning task running multiple times, which is the situation that happened when multiple images were handed off in a short period of time.  In order ensure the fileQueue only retains one entry;  the race condition bug caused multiple entries of the same file to be queued.    I'll add an addition comment.

Reply:
Fixed